osu.Game/Overlays/FirstRunSetup/ScreenImportFromStable.cs

```diff
@@ -122,19 +122,36 @@ private void updateStablePath()
             stableLocatorTextBox.Current.Value = storage.GetFullPath(string.Empty);
             importButton.Enabled.Value = true;
 
-            bool available = legacyImportManager.CheckHardLinkAvailability();
-            Logger.Log($"Hard link support is {available}");
+            bool beatmapsAvailable = legacyImportManager.CheckHardLinkAvailability(StableContent.Beatmaps);
+            bool othersAvailable = legacyImportManager.CheckHardLinkAvailability(StableContent.Scores); // all others use install location
+            Logger.Log($"Hard link support availability - beatmaps:{beatmapsAvailable} others:{othersAvailable}");
 
-            if (available)
+            if (beatmapsAvailable && othersAvailable) // hard links will be use, link to wiki
             {
                 copyInformation.Text =
                     "Data migration will use \"hard links\". No extra disk space will be used, and you can delete either data folder at any point without affecting the other installation. ";
 
                 copyInformation.AddLink("Learn more about how \"hard links\" work", LinkAction.OpenWiki, @"Client/Release_stream/Lazer/File_storage#via-hard-links");
             }
-            else if (RuntimeInfo.OS != RuntimeInfo.Platform.Windows)
-                copyInformation.Text = "Lightweight linking of files is not supported on your operating system yet, so a copy of all files will be made during import.";
-            else
+            else if (beatmapsAvailable) // hard links will be used for beatmaps which the user has cared enough about to manually edit stable .cfg file. Link them to wiki
+            {
+                copyInformation.Text =
+                    "A second copy of scores, skins, and collections will be made during import. Migrating beatmaps will use \"hard links\" thus no extra disk space will be used for them, and you can delete beatmaps at any point without affecting the other installation. ";
+
+                copyInformation.AddLink("Learn more about how \"hard links\" work", LinkAction.OpenWiki, @"Client/Release_stream/Lazer/File_storage#via-hard-links");
+            }
+            else if (othersAvailable) // hard links can not be used for beatmaps, allow the user to change lazer storage location
+            {
+                copyInformation.Text =
+                    "A second copy of beatmaps will be made during import. To avoid this, please make sure the lazer data folder is on the same drive where your previous osu! install stores its beatmaps (and the file system is NTFS). You can find the `BeatmapDirectory` for the previous install in the .cfg file. ";
+                copyInformation.AddLink(GeneralSettingsStrings.ChangeFolderLocation, () =>
+                {
+                    game?.PerformFromScreen(menu => menu.Push(new MigrationSelectScreen()));
+                });
+            }
```

Comment:
Can we not split hairs like this? Can we just stick to all-or-nothing on this? Not only is this difficult for us to maintain, people will have no idea what they're being told here.

Reply:
Sure, but then #21904 has no reason to remain open...

Reply:
Seeing the complexity associated with hypercorrect handling of this I say we just check both paths for hard link availability and tell the user their data is gonna get copied unless they move everything on one drive.

Reply:
So change first if to `if (beatmapsAvailable && othersAvailable)` and keep everything else as is in master?

Reply:
That's what I'd do. Dunno. Need more opinions. I thought this was initially going to be simpler to support well but hadn't considered the complexity behind it.

Reply:
Agree with making sure both the stable folder and the beatmaps folder are in the same drive as the lazer folder. Having one side of lazer data storage copied from stable while the other being linked feels bad.

Reply:
This isn't what's happening here nor anywhere else, this here is only changing the explanation text. Hard links will be used with any imported file where hard linking succeeds via `TryCreateHardLink`. Lazer storage will always be in good state because hard links have to be in the same drive where lazer storage is and every other file will fail to create hard link and be copied instead. As such I don't see copying some files while hard linking other as an issue.

Reply:
The point is that *we don't want to be showing complex explanations to the user*, because the more complicated they are, the less likely it is that they'll understand. *At most* I'd agree to a message saying "your beatmaps are on another drive than the rest of the files, so we can't hard link everything", rather than going into painstaking detail about what is where.

Reply:
Should there be three explanations then? The current ones where either all or none of the files can use hard links, and a new one when stable uses a `BeatmapDirectory` which makes it impossible to hard link all files. The third explanation could then just be that - "Your previous osu! install stores files in multiple drives. Data migration can't make use of hard links for all of them." and then maybe link them to https://osu.ppy.sh/wiki/en/Client/Program_files/User_configuration_file#not-changeable-in-game wiki page in addition to the lazer hard link one??

People who are going to manually edit their .cfg files are a minority anyway. I agree that they can figure out what's going to get hard linked vs what isn't.

Reply:
Not sure linking the wiki page adds anything to the message, but otherwise this reads pretty clear to me.

Reply:
My only concern is that changing the `BeatmapDirectory` is a set and forget kinda deal and users might need a reminder of where their beatmaps are stored.